ui/page/show/view.jsx

```diff
@@ -138,15 +138,20 @@ export default function ShowPage(props: Props) {
   useEffect(() => {
     if (canonicalUrl) {
       const urlPath = pathname + hash;
-      const fullParams =
-        urlPath.indexOf('?') > 0 ? urlPath.substring(urlPath.indexOf('?')) : search.length > 0 ? search : '';
-      const canonicalUrlPath = '/' + canonicalUrl.replace(/^lbry:\/\//, '').replace(/#/g, ':') + fullParams;
+      const path = urlPath.slice(1).replace(/:/g, '#');
+      // in case query is not part of search, it will be identified as queryString and
+      // passed along on replaceState
+      let queryString;
+      try {
+        ({ queryString } = parseURI(path));
+      } catch (e) {}
+      const canonicalUrlPath = '/' + canonicalUrl.replace(/^lbry:\/\//, '').replace(/#/g, ':');
```

Comment:
Saw this yesterday but don't quite understand it. Any example of "query is not part of search" to understand it better?

When feeding into `parseURI`, do we need to replace?  (thought it handles both styles).
Asking just because there are 2 known cases where an actual hash will be there, although probably doesn't matter if you are just looking for `queryString`:

1.  Section hashes.  Used to support in Discover, but probably should continue support since it is standard url stuff. e.g. http://localhost:9090/@Odysee:8#livestream_section
2. Oauth stuff (see initial screenshot).

The new code would have replaced those valid hashes (although again, probably doesn't matter? -- not sure).  The original code only performs the replacement over the `canonical_url` and not the entire path, which seems safer/correct.

Will ship after confirming this.

Reply:
Ok, confirmed this is problematic, as the original issue that was impacting oauth came back.  Reverting this specific change fixes it.

Reply:
Ok, confirmed this is problematic, as the original issue that was impacting oauth came back.  Reverting this specific change fixes it

Reply:
Hm ... I'm confused now ... none of the previous commits solved the original oauth problem. Maybe I got the branch wrong during testing that time.

Reproduce:
- `ip.oidc` branch
- Load http://localhost:9090/@Odysee:8/comicbooks:9
- Refresh.  The url stays correct, but the rendered page is one level above.

Reply:
Sorry @saltrafael, forgot to mention that you need to log in first.  Maybe use a temp gmail account to login via oauth in order to not mess up your stuff, since this branch isn't finalized yet.

Reply:
Let me know if it's better now @infinite-persistence, the problem was that when there's a URI like `http://localhost:9090/@fireship:6/redwood-in-100-seconds#3?t=30` the location "hash" is `3?t=30` (everything after `#`) so parseURI had to strip that as well to get the right uri

Reply:
Yay, I'll give it a try tonight or tomorrow.